tachyon/math/base/semigroups.h

```diff
@@ -380,6 +381,76 @@ class AdditiveSemigroup {
     }
   }
 
+  // Linear combination
+  // - forward: a₀ * rⁿ⁻¹ + a₁ * rⁿ⁻² + ... + aₙ₋₁
+  // - backward: a₀ + a₁ * r + ... + aₙ₋₁ * rⁿ⁻¹
+  // NOTE(chokobole): For a performance reason, I would recommend the
+  // |LinearCombinationInPlace()| if possible.
+  template <bool Forward, typename Container, typename T,
+            typename ReturnTy =
+                typename internal::AdditiveSemigroupTraits<G>::ReturnTy>
+  constexpr static ReturnTy LinearCombination(const Container& values,
```

Comment:
Unlike LinearCombinationInPlace, did you intend to return Zero when size of values is 0?

Reply:
Yes, I believe `LinearCombination()` should do like that and `LinearCombinationInPlace()` can't return anything if empty arrays are given.